internal/report/json/json.go

```diff
@@ -54,6 +55,22 @@ func (r *Reporter) Render() {
 	}
 }
 
+// Artifacts should exclusively contain downloaded artifacts.
+// Therefore, we need to sanitize and remove any intermediate data entries
+// prior to generating the final JSON report.
```

Comment:
Go's docs conventionally start with the method name.

```suggestion
// cleanup removes any information that isn't relevant in the rendered report. Particularly when it comes to
// artifacts, this reporter is only interested in those that have been persisted to the file system.
```